### DISCUSSION TITLE
开了很多标签页想一个个播放 许愿一个播放完毕自动关闭

### DISCUSSION THREAD

**Discussion by ts8zs** [the1812/Bilibili-Evolved#2413] [功能建议 / Ideas]
许愿一个播放完毕自动关闭

**Comment by Geeyun-JY3** [comment]
其实为啥不加入稍后再看然后播放全部，看完再清除。

**Comment by ts8zs** [comment]
有道理 那能直接播放全部动态吗 我是直接用linkclump开了所有

**Reply by Geeyun-JY3** [reply]
目前应该是不能的。
将动态所有视频投稿添加至稍后再看更像是个需求。
~~不知道自定义顶栏能不能实现这个需求。~~
全部动态不可能，总得给个范围吧。

**Reply by ts8zs** [reply]
按顺序咯...

**Reply by Geeyun-JY3** [reply]
按顺序添加多少个呢？

**Reply by ts8zs** [reply]
我是指播放全部动态 不是放入稍后再看...

**Reply by Geeyun-JY3** [reply]
不放进稍后再看，是想直接生成一个播放列表？

**Reply by ts8zs** [reply]
就是按顺序播放全部动态（至最后未看？） 至于怎么实现都行。。。

**Reply by the1812** [reply]
最后总得放在某个播放列表的, 我觉得还不如做个播放完毕自动关闭出来

**Reply by ts8zs** [reply]
许个愿而已 不好实现无视就行了...
主要是播放完毕自动关闭然后自动播放是我自己能实现的最简单的方式...

**Reply by Geeyun-JY3** [reply]
> 就是按顺序播放全部动态（至最后未看？） 至于怎么实现都行。。。

感觉你这需求不通过稍后再看也得整个播放列表出来，但现在的稍后再看就是播放列表（看看 UI、URL 就知道了）。至最后未看怎么判断这个事看自定义顶栏现状，好像还可以实现？我觉得主要问题还是播放列表。B 站这个播放列表现在也就比以前好用了一点，统一了稍后再看和收藏夹的播放，出现的情景多了一点，创建的限制还是挺多的，而且不是用完即走。

> 最后总得放在某个播放列表的, 我觉得还不如做个播放完毕自动关闭出来

内存够的话缺确实无所谓🌚

**Reply by ts8zs** [reply]
话说我感觉这应该提给B站的需求...

**Reply by Geeyun-JY3** [reply]
> 话说我感觉这应该提给B站的需求...

稍后再看做了再做这个可能就动力不大了。就是你得多点几次……

话说这个需求是不是也有另一个实现思路：自动将未看视频添加至稍后再看（前提是能判断哪些视频未看；顶栏的数字的实现原理类似？）。

**Comment by ts8zs** [comment]
```
// ==UserScript==
// @name         B站视频结束自动关闭标签页
// @version      0.1
// @description  Automatically close the current tab when a Bilibili video ends.
// @author       Ts8zs chat with bing
// @match        https://www.bilibili.com/video/*
// @grant        none
// ==/UserScript==
(function() {
    'use strict';

    var video = document.querySelector('video');
    video.addEventListener('ended', function() {
        window.close();
    });
})();
````
